addons/stock/models/res_config_settings.py

```diff
@@ -16,6 +16,10 @@ class ResConfigSettings(models.TransientModel):
         help="Track following dates on lots & serial numbers: best before, removal, end of life, alert. \n Such dates are set automatically at lot/serial number creation based on values set on the product (in days).")
     group_stock_production_lot = fields.Boolean("Lots & Serial Numbers",
         implied_group='stock.group_production_lot')
+    group_lot_on_delivery_slip = fields.Boolean("Display Lots & Serial Numbers", 
```

Comment:
Don't forget to add the groups on the res_users models.

See: https://github.com/odoo/odoo/blob/master/addons/stock/models/res_users.py#L10

You should also add them on the res_users view.